rascal-vscode-extension/src/dap/RascalDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,56 @@
+import {CancellationToken, DebugConfiguration, DebugConfigurationProvider, ProviderResult, WorkspaceFolder, window} from "vscode";
+import { RascalDebugClient } from "./RascalDebugClient";
+
+/**
+ * Class used by Vscode when starting debug sessions to add configuration information before it launchs
+ */
+export class RascalDebugConfigurationProvider implements DebugConfigurationProvider {
+
+    debugClient: RascalDebugClient;
+
+    constructor(debugClient: RascalDebugClient){
+        this.debugClient = debugClient;
+    }
+
+    provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken | undefined): ProviderResult<DebugConfiguration[]> {
+        const conf: DebugConfiguration = {type: "rascalmpl", name: "Rascal Debug", request: "attach"};
+        return [conf];
+    }
+
+    async resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DebugConfiguration, token?: CancellationToken | undefined): Promise<DebugConfiguration> {
+        return new Promise((resolve, reject) => {
+            if(token !== undefined){
+                token.onCancellationRequested((e) => {
+                    reject(e);
+                });
+            }
+
+            if (debugConfiguration.type === undefined) {
+                debugConfiguration.type = "rascalmpl";
+                debugConfiguration.name = "Rascal Debug";
+                debugConfiguration.request = "attach";
+            }
+
+            if (!debugConfiguration.serverPort){
+                window.activeTerminal?.processId.then((value: number|undefined) => {
```

Comment:
can this function be rewritten to use `async` & `await`? Since that already makes it into a promise? and will mean less  `then`/`reject`/`resolve`?